cupy/testing/helper.py

```diff
@@ -101,8 +101,13 @@ def test_func(self, *args, **kw):
             # values are negative.
             if _contains_signed_and_unsigned(kw):
                 inds = _make_positive_indices(self, impl, args, kw)
-                cupy_result = cupy.asnumpy(cupy_result)[inds]
-                numpy_result = cupy.asnumpy(numpy_result)[inds]
+                try:
```

Comment:
My fault, I was not able to understand the whole context.
Is this assumption above the code correct:
> When a input dtype is a unsigned, then negative values in the out array are ignored?

When this is the case both implementations are wrong:
I have only testcases, where the output is float{32, 64}, therefore I never need to skip the equality test. 
You have the case, that a input and the output is unsigned, therefore you correct assume, that you must ignore unsigned values.